src/Nethermind/Nethermind.Merge.Plugin/Synchronization/BeaconPivot.cs

```diff
@@ -91,6 +93,10 @@ public void EnsurePivot(BlockHeader? blockHeader)
                     return;
                 }
 
+                // BeaconHeaderSync actually starts from the parent of the pivot. So we need to to manually insert
+                // the pivot itself here.
+                _blockTree.Insert(blockHeader,
```

Comment:
I thought that it is added in TryInsertDanglingBlock. Isn't it?

Reply:
Not if beacon header sync is not started yet.